shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -71,16 +79,43 @@ public void run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     AlluxioURI srcPath = new AlluxioURI(args[0]);
     AlluxioURI dstPath = new AlluxioURI(args[1]);
-    List<AlluxioURI> srcPaths = AlluxioShellUtils.getAlluxioURIs(mFileSystem, srcPath);
-    if (srcPaths.size() == 0) {
-      throw new FileDoesNotExistException(
-          ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(srcPath.getPath()));
-    }
-
-    if (srcPath.containsWildcard()) {
-      copyWildcard(srcPaths, dstPath, cl.hasOption("R"));
+    if ((dstPath.getScheme() == null || "alluxio".equals(dstPath.getScheme()))
+        && "file".equals(srcPath.getScheme())) {
+      List<File> srcFiles = AlluxioShellUtils.getFiles(srcPath.getPath());
+      if (srcFiles.size() == 0) {
+        throw new IOException(ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(srcPath));
+      }
+      if (srcPath.containsWildcard()) {
+        copyFromLocalWildcard(srcFiles, dstPath);
+      } else {
+        copyFromLocal(new File(srcPath.getPath()), dstPath);
+      }
+    } else if ((srcPath.getScheme() == null || "alluxio".equals(srcPath.getScheme()))
+        && "file".equals(dstPath.getScheme())) {
+      File dstFile = new File(dstPath.getPath());
+      List<AlluxioURI> srcPaths = AlluxioShellUtils.getAlluxioURIs(mFileSystem, srcPath);
+      if (srcPaths.size() == 0) {
+        throw new IOException(ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(srcPath));
+      }
+      if (srcPath.containsWildcard()) {
+        copyWildcardToLocal(srcPaths, dstFile);
+      } else {
+        copyToLocal(srcPath, dstFile);
+      }
+    } else if ((srcPath.getScheme() == null || "alluxio".equals(srcPath.getScheme()))
+        && (dstPath.getScheme() == null || "alluxio".equals(dstPath.getScheme()))) {
+      List<AlluxioURI> srcPaths = AlluxioShellUtils.getAlluxioURIs(mFileSystem, srcPath);
+      if (srcPaths.size() == 0) {
+        throw new FileDoesNotExistException(
+            ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(srcPath.getPath()));
+      }
+      if (srcPath.containsWildcard()) {
+        copyWildcard(srcPaths, dstPath, cl.hasOption("R"));
+      } else {
+        copy(srcPath, dstPath, cl.hasOption("R"));
+      }
     } else {
-      copy(srcPath, dstPath, cl.hasOption("R"));
+      System.out.println("Scheme must be file or alluxio and only support one file scheme.");
```

Comment:
Also, this should return an exception to indicate the command fails? Otherwise the return value of the command will be succeeded.